src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useRef, useState } from 'react';
+import classnames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  onToggle: (todo: Todo) => void,
+  onUpdate: (todoId: number, title: string) => void,
+  onDelete: (todoId: number) => void,
+};
+
+export const TodoItem: React.FC<Props> = React.memo(({
+  todo,
+  onToggle,
+  onUpdate,
+  onDelete,
+}) => {
+  const [editing, setEditing] = useState<boolean>(false);
+  const [titleEdit, setTitleEdit] = useState<string>(todo.title.trim());
+  const currentInputField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (currentInputField.current) {
+      currentInputField.current.focus();
+    }
+  });
+
+  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      setEditing(false);
+      setTitleEdit(todo.title);
+    }
+
+    if (event.key === 'Enter') {
+      if (titleEdit.trim() === '') {
+        onDelete(todo.id);
+        setTitleEdit(prev => prev.trim());
+      }
+
+      if (titleEdit.trim() !== '' && (titleEdit.trim() !== todo.title.trim())) {
+        onUpdate(todo.id, titleEdit.trim());
+        setTitleEdit(prev => prev.trim());
+      }
+
+      setEditing(false);
+    }
+  };
+
+  const handleInputBlur = () => {
+    if (!titleEdit.trim()) {
+      onDelete(todo.id);
+    }
+
+    if (titleEdit !== todo.title) {
+      onUpdate(todo.id, titleEdit.trim());
```

Comment:
Since you already have trimmed text in `titleEdit`, no need to do it here

Reply:
I removed trimming in titleEdit instead, it might work unexoectedly

Reply:
Then you can create a variable with a trimmed value at the start of the function instead of calling `trim()` a few times. But not critical for now, just FYI.